packages/postcss-merge-rules/src/index.js

```diff
@@ -229,22 +243,18 @@ function partialMerge(first, second) {
   }
 
   const firstDecls = getDecls(first);
-
   // Filter out intersections with later conflicts in First
   intersection = intersection.filter((decl, intersectIndex) => {
     const indexOfDecl = indexOfDeclaration(firstDecls, decl);
     const nextConflictInFirst = firstDecls
       .slice(indexOfDecl + 1)
       .filter((d) => isConflictingProp(d.prop, decl.prop));
-    if (!nextConflictInFirst.length) {
+    if (nextConflictInFirst.length === 0) {
       return true;
     }
     const nextConflictInIntersection = intersection
       .slice(intersectIndex + 1)
       .filter((d) => isConflictingProp(d.prop, decl.prop));
-    if (!nextConflictInIntersection.length) {
```

Comment:
Next condition ensures that `nextConflictInIntersection.length !== 0` since already `nextConflictInFirst.length !== 0`, but  in fact it even does not matter, as long as the number of conflicts is the same we can still hope the merge will work.